checkbox-provider-ce-oem/bin/evdev_touch_test.py

```diff
@@ -0,0 +1,225 @@
+            expect_slot:
+        result = True
+    else:
+        print("\n#### Touch events are not expected")
+
+    return result
+
+
+def capture_events(device, event_queue):
+    for event in device.read_loop():
+        event_queue.put(event)
+
+
+def main():
+    """
+    the scripts would capture all touch events to queue by
+    evdev module in background, and compare the event
+    one by one once received a SYN_REPORT event
+
+    ## For the single touch event
+    option 1:
+        a BTN_TOUCH, ABS_X and ABS_Y is required in a event group
+    option 2:
+        ABS_MT_TRACKING_ID, ABS_MT_POSITION_X and ABS_MT_POSITION_Y
+        is required in a event group, the ABS_MT_SLOT event must not exist
+    option 3:
+        ABS_MT_TRACKING_ID, ABS_MT_POSITION_X, ABS_MT_POSITION_Y
+        and ABS_MT_SLOT is required in a event group,
+        only one ABS_MT_SLOT is received
+        and the value of ABS_MT_SLOT event must be 0
+
+    ## For the multiple touch event
+        ABS_MT_TRACKING_ID, ABS_MT_POSITION_X, ABS_MT_POSITION_Y
+        and ABS_MT_SLOT is required in a event group.
+        the value of ABS_MT_SLOT event must be (tap number - 1)
+    """
+
+    args = register_arguments()
+    due_time = time.time() + args.timeout
+
+    print("\n### Available input devices:")
+    for d in [evdev.InputDevice(path) for path in evdev.list_devices()]:
+        print("- {}".format(d.name))
+        if d.name == args.name:
+            device = d
+            break
+    else:
+        raise SystemExit("Touch device not found!")
+    expected_tap = args.xfingers
+
+    print("\n### Start to capture events from {}".format(d.name))
+    events = []
+    result = False
+    event_queue = queue.SimpleQueue()
+    proc = threading.Thread(
+            target=capture_events,
+            args=(device, event_queue,))
+    proc.daemon = True
+    proc.start()
+
+    while True:
+        if time.time() > due_time:
+            d.close()
+            raise TimeoutError("Event not detected")
+
+        if event_queue.empty():
+            continue
+
+        event = event_queue.get_nowait()
+        if event.type == EV_SYN and event.code == SYN_REPORT:
+            detect_state, filter_events = filter_touchscreen_event(events)
+            if expected_tap == 1:
+                result = check_single_touch_event(detect_state)
+            else:
+                result = check_multiple_touch_event(detect_state, expected_tap)
+
+            # Dump touch events
+            for ev in filter_events:
+                print_event(ev)
+
+            if result:
+                print(
+                    ("\n#### {} touch event "
+                     "detected seccessfully").format(expected_tap)
+                )
+                break
+            events = []
+        else:
+            events.append(event)
+    print("### Stop to capture events from {}".format(d.name))
+
+
+if __name__ == "__main__":
+    global events
```

Comment:
Thanks for adding the comments based on our offline talk.
Regarding this global variable, do we still need it?

Reply:
Nope, we don't need this global events variable
I have pushed a commit for this MR, thanks.